python/mpi_cmake_modules/clang_format.py

```diff
@@ -64,24 +60,35 @@ def list_of_files_to_format(files_or_directories):
 
 def execute_clang_format(clang_format_bin, clang_format_config,
                          clang_format_arg):
-    cmd = (clang_format_bin + " -style=\"" + clang_format_config + "\" -i " +
-           ' '.join(clang_format_arg))
-    print ("executing: ", cmd)
-    ret = subprocess.call([cmd])
-    print(ret)
+    cmd = ' '.join([clang_format_bin,
+          ' -style="' + clang_format_config + '"',
+          ' -i ' + ' '.join(clang_format_arg)
+          ])
+    try:
+        print ("executing: ")
+        print (cmd)
+        os.system(cmd)
+    except Exception as e:
+        print("Fail to call " + clang_format_bin + " with error:")
+        print(e)
 
 
 if __name__ == "__main__":
+    """ Format source files given or found recursively in the given folders """
+
+    parser = argparse.ArgumentParser(description='Process some integers.')
```

Comment:
this has an API and can be used as main. I need to put it as part of the python API in case someone wants to base another executable on it, typically the clang-format-diff or check that can be added.

The fact that it is flagged as executable allow the symlink to be used by the rosrun

Reply:
Thanks for the tips I will try